Add second usage ping for the wallet unlock event

DJAndries opened the pull request:
<!-- Add brave-browser issue bellow that this PR will resolve -->
Resolves brave/brave-browser#20956

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

  0. Set current date to Feb 8, 2022
  1. Install browser with a fresh state
  2. Allow usage ping to occur, activate wallet, wait for second usage ping, close browser
     - confirm first usage ping walletActive=0 (inactive)
     - confirm second usage ping walletActive=7 (daily, weekly, monthly)
  3. Increment the date by one day
  4. Open browser, wait for usage ping, open wallet, wait for second usage ping, close the browser
     - confirm first usage ping walletActive=0 (inactive)
     - confirm second usage ping walletActive=1 (daily)
  5. Increment the date by 7 days
  6. Open browser, wait for usage ping, open wallet, wait for second usage ping, close the browser
     - confirm first usage ping walletActive=0 (inactive)
     - confirm second usage ping walletActive=3 (daily, weekly)
  7. Set date to the first day in the next month
  8. Open browser, wait for usage ping, open wallet, wait for second usage ping, close the browser
     - confirm first usage ping walletActive=0 (no use since last report)
     - confirm second usage ping walletActive=7 (daily, weekly, monthly)

bbondy reviewed (changes requested):
This looks right but there's a problem that will lead to under-reported wallet stats. 
We only send 1 ping at most during a day. So if wallet was enabled after that ping is set, then we wouldn't report a daily ping for the wallet. The wallet param would have already reported not used, even though the wallet is later used.

DJAndries commented:
> This looks right but there's a problem that will lead to under-reported wallet stats. We only send 1 ping at most during a day. So if wallet was enabled after that ping is set, then we wouldn't report a daily ping for the wallet. The wallet param would have already reported not used, even though the wallet is later used.

Changed to send a second usage ping if wallet is opened after the first usage ping. Browser monthly/weekly/daily usage flags will be set to false in second ping.

Also moved the "last wallet unlock" timestamp to the local state so that we can observe wallet usages in all browser profiles.

@bbondy I know that you requested using "time travel" style unit tests, but I found it hard to apply to wallet usage stats specifically. We derive monthly/weekly/daily wallet usage flags by comparing the "last unlock time" to the "last reported unlock time". We do not use the current date to perform any assessment.

DJAndries commented:
I will add a new test plan in the OP comment.

DJAndries commented:
As per internal discussion, we will be focusing on improving P3A reporting. A new PR has been created: brave/brave-core#12233

Will leave this PR open in case we decide to improve upon the usage ping functionality by adding a second usage ping when the user unlocks the wallet.

bbondy commented on components/brave_wallet/browser/brave_wallet_service.cc:
```diff
@@ -23,6 +23,7 @@
 #include "brave/components/brave_wallet/common/hex_utils.h"
 #include "brave/components/brave_wallet/common/value_conversion_utils.h"
 #include "brave/components/brave_wallet/common/web3_provider_constants.h"
+#include "chrome/browser/browser_process.h"
```
This is a layer violation, you can't have `chrome/*` includes in `components/brave_wallet`.

bbondy commented on components/brave_wallet/browser/brave_wallet_utils.cc:
```diff
@@ -25,6 +25,7 @@
 #include "brave/components/brave_wallet/common/hex_utils.h"
 #include "brave/components/brave_wallet/common/value_conversion_utils.h"
 #include "brave/vendor/bip39wally-core-native/include/wally_bip39.h"
+#include "chrome/browser/browser_process.h"
```
Same

DJAndries commented:
Closing, since brave/brave-core#12310 and brave/brave-core#12277 are replacements
